core/envelope-sim/src/main/kotlin/fr/sncf/osrd/envelope_sim/etcs/ETCSBrakingSimulator.kt

```diff
@@ -13,51 +14,97 @@ import fr.sncf.osrd.utils.units.Offset
 interface ETCSBrakingSimulator {
     val context: EnvelopeSimContext
 
-    /** Compute the ETCS braking envelope for each LOA. */
+    /**
+     * Compute the ETCS braking envelope for each LoA and return a new envelope taking said curves
+     * into account.
+     */
     fun addSlowdownBrakingCurves(
         envelope: Envelope,
         limitsOfAuthority: Collection<LimitOfAuthority>
     ): Envelope
 
-    /** Compute the ETCS braking envelope for each EOA. */
+    /**
+     * Compute the ETCS braking envelope for each EoA and return a new envelope taking said curves
+     * into account.
+     */
     fun addStopBrakingCurves(
         envelope: Envelope,
         endsOfAuthority: Collection<EndOfAuthority>
     ): Envelope
+
+    /** Compute the ETCS braking curves for each LoA, ordered by LoA offset. */
+    fun computeSlowdownBrakingCurves(
+        mrsp: Envelope,
+        limitsOfAuthority: Collection<LimitOfAuthority>
+    ): LOAETCSBrakingCurves
+
+    /** Compute the ETCS braking curves for each EoA, ordered by EoA offset. */
+    fun computeStopBrakingCurves(
+        mrsp: Envelope,
+        endsOfAuthority: Collection<EndOfAuthority>
+    ): EOAETCSBrakingCurves
 }
 
+typealias ETCSBrakingCurves = EnumMap<BrakingCurveType, ETCSBrakingCurve?>
+
+typealias LOAETCSBrakingCurves = NavigableMap<LimitOfAuthority, ETCSBrakingCurves>
+
+typealias EOAETCSBrakingCurves = NavigableMap<EndOfAuthority, ETCSBrakingCurves>
```

Comment:
nit: LOA and EOA are probably enough (especially with all the caps)
```suggestion
typealias LOABrakingCurves = NavigableMap<LimitOfAuthority, ETCSBrakingCurves>

typealias EOABrakingCurves = NavigableMap<EndOfAuthority, ETCSBrakingCurves>
```

Reply:
Was wondering the same, was waiting for this comment, perfect, works for me.